src/contexts/Polkawatch/index.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 @paritytech/polkadot-staking-dashboard authors & contributors
+// SPDX-License-Identifier: GPL-3.0-only
+
+import { Configuration, PolkawatchApi } from '@polkawatch/ddp-client';
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { useApi } from '../Api';
+import type { NetworkName } from '../../types';
+
+/**
+ * This is the Polkawatch API provider, which builds polkawatch API depending on the Chain that is currently
+ * in context. Also returns information about whether there exist decentralization analytics for the Network.
+ */
+
+const POLKAWATCH_API_VERSION = 'v2';
+const POLKAWATCH_NETWORKS = ['polkadot', 'kusama'];
+
+/**
+ * Builds the API Configuration based on Chain and API Version
+ * @param name the chain to query: polkadot, kusama, etc.
+ * @param version the API version
+ * @constructor
+ */
+const apiConfiguration = (
+  name: NetworkName = 'polkadot',
```

Comment:
This initial network name may be different if the user has a different initial network set in local storage. 

```suggestion

  name: NetworkName = localStorageOrDefault('network', 'polkadot');
```

Reply:
that is a good one, would have messed up things, potentially. implemented.